woocommerce/woocommerce-ios
# Variations: Create all variations from no attributes empty state screen.

Closes: #8534

# Why

This PR handles the case of generating all variations from a no variations and no attributes state. Giving completion to https://github.com/woocommerce/woocommerce-ios/pull/8614

# How 

- Extract the generate all variations code from `ProductVariationsViewModel` to `GenerateAllVariationsUseCase`. All tests were moved too.

- Move the `updateProductVariationCount` code from `ProductVariationsViewController` to the use case.

- Update the edit attribute module to use the `GenerateVariationsOptionsPresenter `, `GenerateAllVariationsPresenter `, and `GenerateAllVariationsUseCase`

- Make sure the `onVariationCreation` closure from the edit attribute module is invoked. Here I removed an unused `ProductVariation` parameter.

# Demo

https://user-images.githubusercontent.com/562080/211963286-bc216f9d-7977-450a-ba72-7aeb0b18cfa5.mov


# Testing Steps

- Create a new variable product
- Tap the variations row
- Start by adding some attributes from the empty state screen
- Tap next to create your attribute.
- Tap the "Generate Variation" button.
- Start the generate all variations process.
- See that all variations are generated correctly.

---
